### PR TITLE
Refactoring of the basic client module

### PR DESCRIPTION
**Pull request by hendrikebbers**
- final
- null checks
- comment code removed
- TODOs that should be discussed

### CODE REVIEW

**Comment by hendrikebbers** [comment]
Maybe this has some duplicate changes with #23 that should be merged before this PR is merged. #23 can be merged once @joha78 reacted in then small review requests I did earlier today.

**Comment by madmas** [comment]
Overall, looks good to me and I agree that #23 needs to be merged first.*
*But* please  remove all Todo comments from the code before merge, either we should solve them within the PR or transfer each topic into an issue

File: base/client/src/main/java/dev/rico/internal/client/projector/JavaFXProjectorImpl.java [hunk 1]
```diff
@@ -83,29 +88,32 @@ private void updateUiRoot(final ItemModel itemModel) {
...
     }
 
-    public SimpleObjectProperty<Node> rootProperty() {
+    public ObjectProperty<Node> rootProperty() {
         return root;
     }
 
     @Override
     public <N extends Node> N createNode(final ItemModel itemModel) {
+        //TODO: Discussion if this really makes sense...
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/JavaFXProjectorImpl.java]: the return null is for this case:
```
BorderPaneModel borderPaneModel = ui().borderPane();
        borderPaneModel.setCenter(something);
        borderPaneModel.setCenter(null);
```

File: base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/DialogConfiguration.java [hunk 2]
```diff
@@ -11,16 +12,20 @@
...
+        final Optional<Node> nodeOptional = Optional.ofNullable(model.getOwner()).map(projector.getModelToNodeMap()::get);
+        final Optional<Window> windowOptional = nodeOptional.flatMap(node -> Optional.of(node).map(Node::getScene).map(Scene::getWindow));
         windowOptional.ifPresent(alert::initOwner);
         alert.setTitle(model.getTitle());
         return nodeOptional;
     }
 
-    default Optional<Window> findWindowOptional(Projector projector, DialogModel newDialog) {
-        Optional<Node> nodeOptional = Optional.ofNullable(newDialog.getOwner()).map(projector.getModelToNodeMap()::get);
+    //TODO: Method not referenced. Can we remove it?
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/DialogConfiguration.java]: Can be removed

File: base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/ShutdownDialogHandler.java [hunk 3]
```diff
@@ -5,12 +5,13 @@
 import dev.rico.internal.projector.ui.dialog.ShutdownDialogModel;
 import javafx.scene.control.Alert;
 
-public class ShutdownDialogHandler implements ProjectorDialogHandler<ShutdownDialogModel>, DialogConfiguration  {
+//TODO: How should we handle this one?
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/ShutdownDialogHandler.java]: will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/mixed/Configuration.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package dev.rico.internal.client.projector.mixed;
 
+//TODO: Class is not used, can we delete it?
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/mixed/Configuration.java]: Will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/uimanager/AsyncSequence.java [hunk 5]
```diff
@@ -9,30 +9,34 @@
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
+//TODO: This class is not used. Can we delete it?
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/uimanager/AsyncSequence.java]: Will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/uimanager/ImageViewer.java [hunk 6]
```diff
@@ -10,85 +11,40 @@
 import javafx.scene.image.Image;
 import javafx.scene.image.ImageView;
 
+//TODO: Looks like a factory / presenter for this class is missing. Should we remove it?
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/uimanager/ImageViewer.java]: Will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/uimanager/UnexpectedErrorDialog.java [hunk 7]
```diff
@@ -15,94 +17,82 @@
...
+        setTitle("Unexpected error");
+        setHeaderText("The application reported an unexpected error at runtime!");
+    }
+
+    private void configureAsRecoverable() {
+        getButtonTypes().add(0, shutDownButton);
+        setTitle("Unexpected error");
+        setHeaderText("The application reported an unexpected error at runtime!");
+    }
 }
```
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/uimanager/UnexpectedErrorDialog.java:98]: Messages should be configurable

**Review by joha78 — CHANGES_REQUESTED** [review]

**Review by madmas — CHANGES_REQUESTED** [review]
* remove TODO (convert into issues or solve)
* solve open discussions 

;-)

**Pull request merged**